### DISCUSSION TITLE
non rectangular buttons

### DISCUSSION THREAD

**Discussion by mobiusstrip** [godotengine/godot-proposals#6189] [GUI]
### Describe the project you are working on

ios puzzle game

### Describe the problem or limitation you are having in your project

i am using isometric maps and need non rectangular buttons 
![Screenshot (526)](https://user-images.githubusercontent.com/69325829/215259751-6db5f344-38bc-4d0f-be4b-657cfbc9d21c.png)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

idk :(

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

idk :(

### If this enhancement will not be used often, can it be worked around with a few lines of script?

idk :(

### Is there a reason why this should be core and not an add-on in the asset library?

im sure it will be useful to a lot of people like what if someone wants a circle button im sorry if this is the wrong place

**Comment by Calinou** [comment]
This can be done with a Button node and a bitmap image that acts as a click mask.

**Reply by ettmetal** [reply]
I know this is a bit old, but this doesn't seem current and I found this looking for exactly this answer. How is this currently done?

**Reply by AThousandShips** [reply]
This is still valid and you can still do that with the click mask

**Reply by ettmetal** [reply]
The Button node in my install of Godot 4.4 has no exported property called "click mask", so I'm not sure what you're referring to.

**Reply by kitbdev** [reply]
TextureButton has a click mask, or you can implement `_has_point`.
